app/controllers/hiring_staff/sessions_controller.rb

```diff
@@ -9,13 +9,11 @@ def new
   def create
     permission = Permission.new(identifier: oid)
 
-    if permission.valid?
-      session.update(session_id: oid)
-      session.update(urn: permission.school_urn)
-      redirect_to school_path(current_school.id)
-    else
-      redirect_to root_path, notice: I18n.t('errors.sign_in.unauthorised')
-    end
+    return redirect_to root_path, notice: I18n.t('errors.sign_in.unauthorised') unless permission.valid?
```

Comment:
Reads nicer having the unhappy case return early leaving the rest of the code to tell the story of the happy path 👍 .